aml/src/resource.rs

```diff
@@ -120,6 +301,147 @@ pub struct IrqDescriptor {
     pub irq: u32,
 }
 
+fn irq_format_descriptor(bytes: &[u8]) -> Result<Resource, AmlError> {
+
```

Comment:
The spec gets a lot of details wrong, I think it's better for our code to do the right thing (the length is `3` or `4` bytes) and just add a comment that the spec is wrong to avoid confusing people in the future.

Nit: stray newline